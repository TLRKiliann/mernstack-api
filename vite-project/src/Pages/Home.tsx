import { Component } from 'react'
import { db_notes } from '../notes/db_notes'
import { notesType } from '../notes/notestype'
//import Header from '../components/Header'
import MainTitle from '../components/MainTitle'
import MainComp from '../components/MainComp'
import Review from '../components/Review'
import Footer from '../components/Footer'
import VoletRight from './Volets/VoletRight'
import VoletLeft from './Volets/VoletLeft'
import '../stylePages/Home.scss'


export default class Home extends Component<notesType> {
  state = {
    notes: [],
    textHeader: "Text Header",
    secondTextHeader: "MY SECOND TEXT HEADER",
    isClosed: false,
    isLefted: false
  }

  componentDidMount() {
    console.log("Mounted !")
    this.setState({notes: db_notes})
  }

  handleVoletsRight = () => {
    console.log(this.state.isClosed)
    this.setState({isClosed: !this.state.isClosed})
  }

  handleVoletsLeft = () => {
    console.log(this.state.isLefted)
    this.setState({isLefted: !this.state.isLefted})
  }

  render() {
    //console.log(this.state.notes)
    return(
      <div>
        <div className="wallIntro">
          <h1>Wellcome To Chat-Room !</h1>
        </div>
        
        <div className="btn--voletsLeft">
          <button onClick={this.handleVoletsLeft}>
            Chat-Room Assistance
          </button>
        </div>
        <div className="btn--voletsRight">
          <button onClick={this.handleVoletsRight}>
            Chat-Room Computing
          </button>
        </div>

        <VoletRight
          isClosed={this.state.isClosed}
          handleVoletsRight={this.handleVoletsRight}
        />

        <VoletLeft 
          isLefted={this.state.isLefted}
          handleVoletsLeft={this.handleVoletsLeft} 
        />

        <MainTitle
          textHeader={this.state.textHeader}
          secondTextHeader={this.state.secondTextHeader}
        />

        <MainComp notes={this.state.notes} />

        <Review results={this.state.notes} />

        <Footer />

      </div>
    )
  }
}