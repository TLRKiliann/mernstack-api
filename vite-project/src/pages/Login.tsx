import React, { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import AuthenticationService from '../services/authentication-service'
import { app } from "../api/axiosconfig";
import { useAuthLogin } from '../context/AuthProvider'
import zoomImg from '../assets/background/circle.jpg'
import '../stylePages/Login.scss'


type Field = {
  value?: any
  error?: string
  isValid?: boolean
}

type Form = {
  username: Field
  password: Field
}

const Login: React.FC = () => {

  const Navigate = useNavigate()
  
  const LOGIN_URL = '/login';
  
  const [form, setForm] = useState<Form>({
    username: {value: ''},
    password: {value: ''}
  })

  const [message, setMessage] = useState<string>('Not connected!');

  const { 
    toggle,
    switchLogin,
    setUserName,
    setAuth
  } = useAuthLogin();

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    const fieldName: string = e.target.name;
    const fieldValue: string = e.target.value;
    const newField: Field = { [fieldName]: { value: fieldValue } };

    setForm({ ...form, ...newField});
  }
  //console.log(form.username, "-- form --")

  const validateForm = () => {
    let newForm: Form = form;

    // Validator username
    if (form.username.value.length < 1) {
      const errorMsg: string = 'Votre pr√©nom doit faire au moins 3 caract√®res de long.';
      const newField: Field = { value: form.username.value, error: errorMsg, isValid: false };
      newForm = { ...newForm, ...{ username: newField } };
    } else {
      const newField: Field = { value: form.username.value, error: '', isValid: true };
      newForm = { ...newForm, ...{ username: newField } };
    }

    // Validator password
    if (form.password.value.length < 1) {
      const errorMsg: string = 'Votre mot de passe doit faire au moins 6 caract√®res de long.';
      const newField: Field = { value: form.password.value, error: errorMsg, isValid: false };
      newForm = { ...newForm, ...{ password: newField } };
    } else {
      const newField: Field = { value: form.password.value, error: '', isValid: true };
      newForm = { ...newForm, ...{ password: newField } };
    }
    setForm(newForm);
    return newForm.username.isValid && newForm.password.isValid;
  }

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const isFormValid = validateForm();
    if (isFormValid) {
      setMessage('üëâ  Tentative de connexion en cours ...');

      AuthenticationService
        .login(form.username.value, form.password.value)
        .then(isAuthenticated => {
          if (!isAuthenticated) {
            setMessage('üîê  Identifiant ou mot de passe incorrect.');
            return;
          } else {
            setAuth(form.username.value, form.password.value)
            setUserName(form.username.value)
            localStorage.setItem("user-info",
              JSON.stringify([form.username.value, form.password.value]))
            toggle()
            console.log("login ok")
            Navigate('/')
          }
        });
    }
  }

  return(
    <div data-testid="logintest" className="bg--loginimg">
      <div className="zoom--img">
        <img
          src={zoomImg}
          width="100%"
          height="100%"
          className="img--login"
          alt={zoomImg}
        />
      </div>

      <div className="div--submit">
        <form
          data-testid="formlog"
          onSubmit={(e) => handleSubmit(e)}
          className="submit"
          placeholder="lastname"
        >
          <h1 className="title--framelogin">Login</h1>

          {message && <div className="form-group">
            <div className="error--message">
              {message}
            </div>
          </div>
          }
          <label htmlFor="username" className="lbl--login">
            Username
          </label>
          <input
            type="text"
            name="username"
            value={form.username.value}
            onChange={(e) => handleInputChange(e)}
            placeholder="username"
            autoComplete="off"
            className="input--login"
            required
          />
          {form.username.error &&
            <div className="error username"> 
             {form.username.error} 
            </div>
          } 
          <label htmlFor="password" className="lbl--login">
            Password
          </label>
          <input
            type="password"
            name="password"
            value={form.password.value}
            onChange={(e) => handleInputChange(e)}
            placeholder="password"
            className="input--login"
            required
          />

          {form.password.error &&
            <div className="error password"> 
              {form.password.error} 
            </div>
          } 
          <button type="submit" className="btn--submitlogin">
            Enter
          </button>

          <p data-testid="linktestlogin" className="p--loginframe">
            <Link
              to="/subscribe"
              className="link--subscribe"
            >
              subscribe
            </Link>
          </p>

        </form>
      </div>

    </div>
  )
}

export default Login;