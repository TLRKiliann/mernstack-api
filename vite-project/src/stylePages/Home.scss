$orangered: #ff4500;
$sandybrown: #f4a460;
$darkmagenta: #8B008B;
$whitesmoke: #f5f5f5;
$lightgrey_16: #161616;
$font_acme: 'Acme', sans-serif;

.div--home {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  min-height: 120vh;
  background: linear-gradient(90deg,
    sandybrown 10%, rgba(195,87,116, 1) 100%);
  .img--home {
    position: fixed;
    top: 0px;
    right: 0px;
    bottom: 0px;
    left: 0px;
    width: 90%;
    height: 70%;
    margin: auto;
    opacity: 0;
    z-index: 1;
  }
  /*.wallIntro {
    position: absolute;
    top: 0px;
    right: 0px;
    left: 0px;
    min-height: 100%;
    background-color: $lightgrey_16;
    overflow: hidden;
    opacity: 0;

    animation-name: wall;
    animation-delay: 0s;
    animation-duration: 3s;
    animation-timing-function: ease;
    .wallIntro--h1 {
      margin-top: 25%;
      padding-top: 20px;
      padding-bottom: 20px;
      text-align: center;
      font-family: $font_acme;
      font-size: 9rem;
      text-shadow: 5px 3px 1px $orangered,
        8px 5px 10px $whitesmoke;
      color: $sandybrown;

      animation-name: main--titleintro;
      animation-delay: 0s;
      animation-duration: 4s;
      animation-timing-function: ease;
      animation-direction: linear;
      animation-iteration-count: 1;
    }
  }*/

  @keyframes main--titleintro {
    0% {
      opacity: 1;
      transform: scale(0);
      z-index: 22;
    }
    40% {
      opacity: 1;
      transform: scale(1.4);
      z-index: 22;
    }
    45% {
      opacity: 1;
      transform: scale(0.8);
      z-index: 22;
    }
    50% {
      opacity: 1;
      transform: scale(1.8);
      z-index: 22;
    }
    100% {
      opacity: 0;
      transform: scale(0);
      z-index: 22;
    }
  }

  @keyframes wall {
    0% {
      opacity: 1;
      z-index: 22;
    }
    70% {
      opacity: 1;
      z-index: 22;
    }
    100% {
      opacity: 0;
      z-index: 22;
    }
  }
  .div--voletsLeft {
    position: absolute;
    margin-top: 12%;
    margin-left: 8%;
    border-radius: 15px;
    z-index: 3;
    .btn--voletLeft {
      padding: 20px 40px;
      font-family: $font_acme;
      font-size: 1.4rem;
      color: rgba(128, 0, 128, 0.4);
      background: linear-gradient(90deg, $sandybrown, rgba(195,87,116, 0.6));
      border: none;
      outline: none;
      border-radius: 40px;
      transition: background 2s, transform 2s;
      &:hover {
        transition: ease;
        transition-duration: 2s;

        background: linear-gradient(90deg, $sandybrown, rgba(195,87,116, 1));

        color: $whitesmoke;
        cursor: pointer;
      }
      &:active {
        transform: scale(0.5);
        transition: ease;
        transition-duration: 2s;
        border: none;
        outline: none;
      }
    }
    .webdevanim {
      margin-top: 250px;
      justify-content: space-between;
      font-family: $font_acme;
      font-size: 1.4rem;
      color: rgba(195,87,116, 1);

      animation-name: rotate-logo;
      animation-delay: 0s;
      animation-duration: 3s;
      animation-iteration-count: infinite;
      animation-timing-function: ease-in-out infinite;
      animation-direction: alternate;
    }
  }
  .div--voletsRight {
    position: absolute;
    right: 0px;
    margin-top: 12%;
    margin-right: 8%;
    border-radius: 15px;
    z-index: 3;
    .btn--voletRight {
      padding: 20px 40px;
      font-family: $font_acme;
      font-size: 1.4rem;
      font-weight: bold;
      color: rgba(128, 0, 128, 0.4);
      background: linear-gradient(90deg,
        rgba(244,164,96, 1) 10%, rgba(195,87,116, 0.6) 100%);
      border: none;
      outline: none;
      border-radius: 40px;
      transition: background 2s, transform 2s;
      &:hover {
        transition: ease;
        transition-duration: 2s;
        background: linear-gradient(90deg,
          rgba(244,164,96, 1) 0%, rgba(195,87,116, 1) 100%);
        color: $whitesmoke;
        cursor: pointer;
      }
      &:active {
        transform: scale(0.5);
        transition: ease;
        transition-duration: 2s;
        border: none;
        outline: none;
      }
    }
    .computinganim {
      margin-top: 250px;
      justify-content: space-between;
      font-family: $font_acme;
      font-size: 1.4rem;
      color: $sandybrown;

      animation-name: rotate-logo-2;
      animation-delay: 0s;
      animation-duration: 8s;
      animation-iteration-count: infinite;
      animation-timing-function: ease-in-out infinite;
      animation-direction: linear;
    }
  }

  @keyframes rotate-logo {
    0% {
      transform: rotate3d(0, 0, 0, 0deg);
    }
    25% {
      transform: rotate3d(3, -2, 3, 360deg);
    }
  }

  @keyframes rotate-logo-2 {
    0% {
      transform: rotate3d(2, -2, -2, 0deg);
    }
    10% {
      transform: rotate3d(3, -2, 3, 90deg);
    }
    20% {
      transform: rotate3d(-1, 2, -1, 180deg);
    }
    30% {
      transform: rotate3d(-3, 0, -3, 90deg);
    }
    40% {
      transform: rotate3d(-3, -2, -2, 180deg);
    }
    50% {
      transform: rotate3d(3, 0, 3, 90deg);
    }
    60% {
      transform: rotate3d(-3, 2, -2, 180deg);
    }
    70% {
      transform: rotate3d(0, -1, 0, 180deg);
    }
    80% {
      transform: rotateX(360deg);
    }
    90% {
      transform: rotateY(0deg);
    }
    100% {
      transform: rotateY(360deg);
    }
  }
}

@media screen and (max-width: 800px) {
  .img--home {
    width: 600px;
    height: 400px;
  }
}

@media screen and (min-width: 360px) and (max-width: 1050px) {
  .div--home {
    background: radial-gradient(800px at 50% 20%, 
      $sandybrown, $darkmagenta);
    .img--home {
    }
    .wallIntro {
    }
    .div--voletsLeft {
      margin-top: 380px;
      margin-left: 12%;
      .btn--voletLeft {
        padding: 10px 20px;
        font-family: $font_acme;
        font-size: 0.9rem;
      }
      .webdevanim {
        display: none;
      }
    }
    .div--voletsRight {
      margin-top: 380px;
      margin-right: 12%;
      .btn--voletRight {
        padding: 10px 20px;
        font-family: $font_acme;
        font-size: 0.9rem;
      }
      .computinganim {
        display: none;
      }
    }
  }
}