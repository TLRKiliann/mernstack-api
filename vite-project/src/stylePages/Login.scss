$royalblue: #4169e1;
$orange: #ffa500;

.bg--loginimg {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  min-height: 100vh;
  background: linear-gradient(180deg, purple, $orange);
  z-index: -1;

  .zoom--img {
    position: fixed;
    top: 0px;
    right: 0px;
    bottom: 0px;
    left: 0px;
    width: 100vw;
    margin: auto;
    opacity: 0.1;
    z-index: -2;
  }

  .img--login {
    width: 100vw;
    min-height: 100%;
    margin: auto;
  }

  .submit {
    padding-bottom: 20px;
    width: 500px;
    margin: auto;
    margin-top: 20%;
    margin-bottom: 20px;
    display: flex;
    flex-direction: column;

    background-color: rgba(45, 85, 255, 0.4);
    box-shadow: 0px 0px 3px orange;
    border-radius: 15px;

    .title--framelogin {
      width: 64%;
      margin: auto;
      margin: 20px auto;
      font-size: 2.2rem;
      color: whitesmoke;
      text-shadow: 2px 2px 1px purple,
      4px 3px 2px #242424;
    }

    .lbl--login {
      width: 64%;
      margin: auto;
      margin-top: 10px;
      font-family: 'Junction';
      font-size: 1.3rem;
      font-weight: bold;
    }

    .input--login {
      width: 60%;
      padding: 3px 10px;
      margin: auto;
      margin-top: 5px;
      font-size: 1.2rem;
      background-color: rgba(85, 45, 255, 0.4);
      border: 1mm inset rgba(150, 45, 255, 0.8);
      color: lightgrey;
    }

    .btn--submitlogin {
      width: 140px;
      margin: auto;
      margin-top: 30px;
      padding: 5px 10px;
      font-size: 1.2rem;
      font-weight: bold;
      background-color: $royalblue;
      border: 3px outset $royalblue;
      border-radius: 12px;
      color: lightgrey;

      &:hover {
        background-color: dodgerblue;
        border: 3px outset dodgerblue;
        cursor: pointer;
        transform: scale(1.1);
        color: lightblue;
      }
      &:active {
        background-color: $royalblue;
        border: 3px outset $royalblue;
        transform: scale(0.9);
        color: white;
      }
    }
    .p--loginframe {
      width: auto;
      margin: auto;
      margin-top: 20px;
      font-size: 1.2rem;
      &:hover {
        transform: scale(1.2);
      }
    }

    .link--subscribe {
      text-decoration: none;
      color: whitesmoke;
      text-shadow: 2px 1px 2px purple;
      &:hover {
        color: $orange;
        text-shadow: 0px 0px 0px purple;
      }
    }

    .error--message {
      width: 60%;
      margin: auto;
      padding: 5px 10px;
      font-size: 0.9rem;
      font-weight: bold;
      color: $orange;
      background-color: #282828;
      border-radius: 5px;
    }

    .error {
      margin: 10px 70px;
      text-align: justify;
    }
  }
}

@media only screen and (max-width: 600px) {
  .submit {
    margin: auto;
    margin-top: 2%;
  }
}

@media only screen and (min-width: 800px) {
  .submit {
    margin: auto;
    margin-top: 20%;
  }
}

@media screen and (min-width: 370px) and (max-width: 412px) {
  .zoom--img {
    width: 400px;
    height: 400px;
    margin: auto;
  }
  .submit {
    margin-top: 60%;
    margin-right: 5%;
    margin-left: 5%;
    font-size: 0.8rem;
  }
}

@media screen and (min-width: 400px) and (max-width: 999px) {
  .submit {
    margin-top: 67%;
  }
}

@media only screen and (min-width: 400px) and (max-width: 1024px) {
  .zoom--img {
    width: 900px;
  }
  .submit {
    margin-top: 240px;
    margin-right: 30%;
    margin-left: 30%;
    font-size: 1.2rem;
  }
}
