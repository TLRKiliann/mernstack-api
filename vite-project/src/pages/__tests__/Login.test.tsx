import React from "react";
import { fireEvent, screen, render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { expect, vi, test } from 'vitest'
import { Link, MemoryRouter } from 'react-router-dom'
import Login from "../Login.tsx";
import handleInputChange from '../Login'
import generateId from '../Login'
import Home from '../Home'
import { PropsWithChildren } from 'react'
//import {assert, assertType, expectTypeOf, beforeEach, afterEach, describe, expect, test, it, vi} from 'vitest';
//import {fireEvent, render, screen} from '@testing-library/react';
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom";
//import { act } from 'react-dom/test-utils';

function wrapper({children}: PropsWithChildren<unknown>) {
  return (
    <MemoryRouter>
      <Login />
    </MemoryRouter>
  )
}

test("should have sign-up link", () => {
  setup()
  expect(screen.getByRole('link', {name: /subscribe/i})).toBeInTheDocument()
})

function setup() {
  render(<Home />, {wrapper})
}

test('handleInputChange to be defined', () => {
  const funcDelete = handleInputChange
  expect(funcDelete).toBeDefined()
})

/*
test("Login subscribe btn test", () => {
  const onClick = vi.fn();
  const { getByTestId } = render(
    <Link
      data-testid="linktestlogin"
      to="/subscribe"
    >
    </Link>
  )
  fireEvent.submit(getByTestId("linktestlogin"));
  expect(onSubmit).toHaveBeenCalledTimes(1);
})
*/

test("Login submit form test", () => {
  const onSubmit = vi.fn();
  const { getByTestId } = render(
    <form
      onSubmit={onSubmit} 
      data-testid="formlog">
    </form>
  )
  fireEvent.submit(getByTestId("formlog"));
  expect(onSubmit).toHaveBeenCalledTimes(1);
})

test('Login generateId to be defined', () => {
  const funcGenIdLog = generateId
  expect(funcGenIdLog).toBeDefined()
})

test('handleInputChange return id', () => {
  const beverage = {
    id: 1,
    img: "http://localhost:5173/src/assets/snapface/oliver.jpg",
    firstName: "tree",
    lastName: "leaf",
    age: 23,
    email: "mail@gtruc.com",
    location: "Valley",
    gender: "male",
    mainroom: "Cyber-Security",
    room: "Scapy",
    isConnected: false,
    signalRecieve: false,
    sentMsg: "hello",
    messagebox: "msg cool",
    returnConfirm: false
  }
  const handleInputChange = vi.fn(beverage => {beverage})
  handleInputChange()
  expect(handleInputChange).toHaveReturned(23)
})

/*
test('renders Login by text', () => {
  render(<Login />);
  const linkElement = screen.getByTestId("logintest");
  expect(linkElement).toBeInTheDocument();
})
*/

/*
describe('<Login />', () => {
  test('The input field and its props', () => {
    render(<Login />)
    const input = document.querySelector(
      'input'
    ) as HTMLInputElement | null;
    
    // input exists in the form component
    expect(input).toBeTruthy()
    
    // is empty
    expect(input?.textContent).toBe('')

    if (input) {
      // test the input text
      input.textContent = 'gaston'
      expect(input.textContent).toBe('gaston')

      // test the type prop
      expect(input.type).toBe('username')
      
      // test the name prop
      expect(input.name).toBe('username')

      // test the value prop
      fireEvent.change(input, {
        target: {
          value: '1234567'
        }
      })
      expect(input.value).toBe('1234567')

      // test the required prop with the jest-dom
      expect(input).toBeRequired()
    }
  })
})
*/