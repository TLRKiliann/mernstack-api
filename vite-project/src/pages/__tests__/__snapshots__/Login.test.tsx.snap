// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<a
  href="/subscribe"
  onClick={[Function]}
>
  Subscribe
</a>
`;
